flume-ng-core/src/test/java/org/apache/flume/source/http/TestHTTPSource.java

```diff
@@ -194,6 +195,14 @@ public void setUp() {
     HttpClientBuilder builder = HttpClientBuilder.create();
     httpClient = builder.build();
     postRequest = new HttpPost("http://0.0.0.0:" + httpPort);
+    SourceCounter sc = (SourceCounter) Whitebox.getInternalState(httpSource, "sourceCounter");
+    sc.start();
+  }
+
+  @After
+  public void tearDown() {
+    SourceCounter sc = (SourceCounter) Whitebox.getInternalState(httpSource, "sourceCounter");
+    sc.stop();
```

Comment:
The reset of the counter is needed to be able to count for each `@Test` method separately.